src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -780,12 +781,18 @@ public static object BindToMoniker(string monikerName)
             return obj;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2050:UnrecognizedReflectionPattern",
+            Justification = "The calling method is guarded")]
```

Comment:
```suggestion
            Justification = "The calling method is annotated with RequiresUnreferencedCode")]
```

Same below - just to be more precise.